packages/react/tsconfig.json

```diff
@@ -1,14 +1,16 @@
 {
   "compilerOptions": {
+    "moduleResolution": "node",
+    "baseUrl": ".",
     "declaration": true,
     "esModuleInterop": true,
     "jsx": "react-jsx",
     "rootDir": "./src",
     "outDir": "./dist",
     "skipLibCheck": true,
     "sourceMap": true,
-    "module": "commonjs",
-    "target": "es5"
+    "module": "esnext",
+    "target": "es2015"
```

Comment:
`target` means the language features:
> https://stackoverflow.com/questions/41993811/understanding-target-and-module-in-tsconfig

Targeting evergreen browsers, this could even be `esnext` but I figure we'll standup more examples (CRA, Nuxt, etc.) to determine what that impact is.

But the less TS has to transpile, the faster our builds will be.